packages/vscode-extension/src/devices/AndroidPhysicalDevice.ts

```diff
@@ -178,6 +210,21 @@ export class PhysicalAndroidDeviceProvider
           available: false,
         })
       );
+
+    const reconnectedDeviceIds = devices
+      .filter((d) => {
+        const prevDevice = previousDevices.find((pd) => pd.id === d.id);
+        return prevDevice && !prevDevice.available;
+      })
+      .map((d) => d.id);
+
+    for (const deviceId of reconnectedDeviceIds) {
+      const activeDevice = this.activeDevices.get(deviceId);
+      if (activeDevice) {
+        activeDevice.notifyDeviceReconnected();
```

Comment:
I don't like this flow.
The way you wrote it, you have:
`DevicesProvider` -(calls)-> `AndroidPhysicalDevice` -(emits event)-> `DeviceSession` -(calls)->`AndroidPhysicalDevice`
The only reason `AndroidPhysicalDevice` is there (and is aware of the whole reconnection mechanism) is so that `DeviceSession` has something to listen on.
Instead, I think it'd be better for the `DeviceSession` to use `DevicesState` -- you can have a state change listener on `DevicesState` that checks if `deviceInfo.available` changed from `false` to `true` (thus, the device reconnected) and trigger the whole flow from there instead.